lib/ros_value.h

```diff
@@ -4,14 +4,17 @@
 #include <cstdint>
 #include <cstring>
 #include <memory>
-#include <pybind11/buffer_info.h>
 #include <string>
 #include <unordered_map>
 #include <vector>
 
 #include "span.hpp"
 #include "util.h"
 
+namespace pybind11 {
+  struct buffer_info;
+}
+
```

Comment:
Hmm forward declarations can be are a dangerous hole to go down - Our style guide says to avoid them whenever possible https://google.github.io/styleguide/cppguide.html#Forward_Declarations

Why do you need the forward declaration here? Can you structure this in a way that doesn't need it?

Reply:
Liam I'll DM you to discuss this a bit, I think I understand the problem but want to dig in just a little more